modules/programs/xplr.nix

```
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+with builtins;
+let
+  inherit (lib)
+    types mkIf mkOption mkEnableOption mkPackageOptionMD mdDoc
+    literalExpression;
+
+  cfg = config.programs.xplr;
+  initialConfig = ''
+    version = '${cfg.package.version}'
+  '';
+  # we provide a default version line within the configuration file, which is obtained from the package's attributes
+  # merge the initial configFile, a mapped list of plugins and then the user defined configuration to obtain the final configuration
+  pluginPath = if cfg.plugins != [ ] then
+    (''
+      package.path=
+    '' + (concatStringsSep " ..\n" (map (p: ''"${p}/init.lua;"'') cfg.plugins))
+      + ''
+         ..
```

```suggestion
      package.path=
    '' + (concatStringsSep " ..\n" (map (p: ''"${p}/init.lua;${p}/?.lua;"'') cfg.plugins))
      + ''
         ..
```

Some plugins can contain multiple Lua packages.

---

@NotAShelf I think you missed this...

---

No, I just can't work with git. History got messed up, again.